src/core/router.js

```diff
@@ -0,0 +1,26 @@
+import { routes } from '../routes';
+
+export function routeRender() {
+  if (!location.hash) {
+    history.replaceState(null, '', '/#/');
+  }
+  const routerView = document.querySelector('router-view');
+  const [hash, queryString] = location.hash.split('?');
+  const currentRoute = routes.find((route) => {
+    return new RegExp(route.path + '/?$').test(hash);
+  });
+  routerView.innerHTML = '';
+  routerView.append(new currentRoute.component().el);
+  window.scrollTo(0, 0);
+}
+
+export const getUrlParam = () => {
+  const [hash, queryString] = location.hash.split('?');
+  const [id, value] = queryString.split('=');
+  return value;
+};
+
+export const navigate = (url = '/#/') => {
+  window.history.pushState(null, null, url);
+  routeRender();
+};
```

Comment:
매번 /#/ 을 붙여줘야 하는 게 번거로운 것 같은데, naviate 함수에서 prefix를 붙여줬다면 어땠을까요?
<img width="419" alt="image" src="https://github.com/KDT1-FE/Y_FE_JAVASCRIPT_PICTURE/assets/57122180/14294c2d-e0ba-417f-8424-d6bc4e3f8a48">

```suggestion
export const navigate = (url) => {
  window.history.pushState(null, null, `/#${url}`);
  routeRender();
};
```

Reply:
앟ㅎㅎㅎ 더 좋은 방법이 있었네요.. 수정했습니다 감사합니다~!
https://github.com/KDT1-FE/Y_FE_JAVASCRIPT_PICTURE/pull/7/commits/b8aa5b2ca7d48c567c80c62165fbeeb5eb6fbfd0